pkg/kubelet/dockertools/manager.go

```diff
@@ -560,7 +569,10 @@ func (dm *DockerManager) runContainer(
 	}
 	namespacedName := types.NamespacedName{pod.Namespace, pod.Name}
 	labels := map[string]string{
-		"io.kubernetes.pod.name": namespacedName.String(),
+		kubernetesNameLabel: namespacedName.String(),
+	}
+	if pod.Spec.TerminationGracePeriodSeconds != nil {
+		labels[kubernetesTerminationGracePeriodLabel] = strconv.FormatInt(*pod.Spec.TerminationGracePeriodSeconds, 10)
```

Comment:
> On May 22, 2015, at 3:30 PM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/kubelet/dockertools/manager.go:
> 
> > @@ -560,7 +569,10 @@ func (dm *DockerManager) runContainer(
> >  	}
> >  	namespacedName := types.NamespacedName{pod.Namespace, pod.Name}
> >  	labels := map[string]string{
> > -		"io.kubernetes.pod.name": namespacedName.String(),
> > +		kubernetesNameLabel: namespacedName.String(),
> > +	}
> > +	if pod.Spec.TerminationGracePeriodSeconds != nil {
> > +		labels[kubernetesTerminationGracePeriodLabel] = strconv.FormatInt(*pod.Spec.TerminationGracePeriodSeconds, 10)
> What is the reason to store the termination grace period as a docker label?
> 
It's to defend if the kubelet is restarted and cannot find the pod - so that we guarantee that we don't super aggressively kill the pod.  I refactored the kubelet to ensure we pass the most current pod everywhere we can.
> What will happen if the user updates the termination grace period?
> 
An update to the pod (I believe) would not restart the containers, but would not update the label.  I don't think we can change labels without a container restart, but I can check.  Tempted to say "requires a restart to get a longer grace period".
> —
> Reply to this email directly or view it on GitHub.
>